code/game/mecha/equipment/tools/mining_tools.dm

```diff
@@ -108,6 +108,18 @@
 /obj/item/mecha_parts/mecha_equipment/mining_scanner/New()
 	SSobj.processing |= src
 
+/obj/item/mecha_parts/mecha_equipment/mining_scanner/attach(obj/mecha/M)
+	..()
+	M.occupant_sight_flags |= SEE_TURFS
+	if(M.occupant)
+		M.occupant.update_sight()
```

Comment:
I changed the effect of the mech mining scanner, it now gives you meson sight constantly instead of switching back and forth. It makes sight updates easier.